Error Messages when i try to compile the code

Hi,

i got all the errors when i try to compile the code.

Arduino: 1.6.8 (Windows 7), Board: "Arduino/Genuino Uno"

E:\Downloads\EggDuino\Functions.ino: In function 'void queryPen()':

Functions:29: error: 'sendAck' was not declared in this scope

  sendAck();

```
      ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void queryButton()':

Functions:34: error: 'sendAck' was not declared in this scope

  sendAck();

```
      ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void queryLayer()':

Functions:40: error: 'sendAck' was not declared in this scope

  sendAck();

```
      ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void setLayer()':

Functions:50: error: 'sendAck' was not declared in this scope

   sendAck();

```
       ^
```

Functions:53: error: 'sendError' was not declared in this scope

   sendError();

```
         ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void queryNodeCount()':

Functions:58: error: 'sendAck' was not declared in this scope

  sendAck();

```
      ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void setNodeCount()':

Functions:69: error: 'sendAck' was not declared in this scope

   sendAck();

```
       ^
```

Functions:72: error: 'sendError' was not declared in this scope

   sendError();

```
         ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void nodeCountIncrement()':

Functions:77: error: 'sendAck' was not declared in this scope

  sendAck(); 

```
      ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void nodeCountDecrement()':

Functions:82: error: 'sendAck' was not declared in this scope

  sendAck();

```
      ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void stepperMove()':

Functions:93: error: 'sendError' was not declared in this scope

   sendError();

```
         ^
```

Functions:97: error: 'sendAck' was not declared in this scope

  sendAck();

```
      ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void setPen()':

Functions:129: error: 'sendError' was not declared in this scope

```
 sendError();

           ^
```

Functions:136: error: 'sendAck' was not declared in this scope

   sendAck();

```
       ^
```

Functions:140: error: 'sendAck' was not declared in this scope

   sendAck();

```
       ^
```

Functions:145: error: 'sendError' was not declared in this scope

   sendError();

```
         ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void togglePen()':

Functions:161: error: 'sendAck' was not declared in this scope

  sendAck();

```
      ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void enableMotors()':

Functions:190: error: 'sendAck' was not declared in this scope

```
 sendAck();

         ^
```

Functions:196: error: 'sendError' was not declared in this scope

```
 sendError();

           ^
```

Functions:203: error: 'sendAck' was not declared in this scope

```
 sendAck();

         ^
```

Functions:209: error: 'sendError' was not declared in this scope

```
 sendError();

           ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void stepperModeConfigure()':

Functions:228: error: 'storePenDownPosInEE' was not declared in this scope

```
 storePenDownPosInEE();

                     ^
```

Functions:229: error: 'sendAck' was not declared in this scope

```
 sendAck();

         ^
```

Functions:232: error: 'storePenUpPosInEE' was not declared in this scope

```
 storePenUpPosInEE();

                   ^
```

Functions:248: error: 'sendError' was not declared in this scope

```
  sendError();

            ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void unrecognized(const char*)':

Functions:259: error: 'sendError' was not declared in this scope

  sendError();

```
        ^
```

E:\Downloads\EggDuino\Functions.ino: In function 'void ignore()':

Functions:263: error: 'sendAck' was not declared in this scope

  sendAck();

```
      ^
```

E:\Downloads\EggDuino\Helper_Functions.ino: In function 'void initHardware()':

Helper_Functions:5: error: 'loadPenPosFromEE' was not declared in this scope

  loadPenPosFromEE();

```
               ^
```

exit status 1
'sendAck' was not declared in this scope

Thanks for it Yura, now the following Errors appear
Do I something wrong or do i have not the right Arduino IDE repositories missing ?
Arduino: 1.6.9 Hourly Build 2016/04/08 08:33 (Windows 10), Board: "Arduino/Genuino Uno"

sketch\EggDuino.ino.cpp.o: In function `setup':

C:\3D Printer\eggbot one piece\eggbotaktuell\EggDuino-master\EggDuino/EggDuino.ino:131: undefined reference to `makeComInterface()'

C:\3D Printer\eggbot one piece\eggbotaktuell\EggDuino-master\EggDuino/EggDuino.ino:132: undefined reference to `initHardware()'

sketch\EggDuino.ino.cpp.o: In function `loop':

C:\3D Printer\eggbot one piece\eggbotaktuell\EggDuino-master\EggDuino/EggDuino.ino:136: undefined reference to `moveOneStep()'

collect2.exe: error: ld returned 1 exit status

exit status 1
Error compiling for board Arduino/Genuino Uno.

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

Here the report with more information:
Arduino: 1.6.9 Hourly Build 2016/04/08 08:33 (Windows 10), Board: "Arduino/Genuino Uno"

C:\3D Printer\eggbot one piece\arduino-nightly\arduino-builder -dump-prefs -logger=machine -hardware "C:\3D Printer\eggbot one piece\arduino-nightly\hardware" -hardware "C:\Users\Daniel\AppData\Local\Arduino15\packages" -tools "C:\3D Printer\eggbot one piece\arduino-nightly\tools-builder" -tools "C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr" -tools "C:\Users\Daniel\AppData\Local\Arduino15\packages" -built-in-libraries "C:\3D Printer\eggbot one piece\arduino-nightly\libraries" -libraries "C:\Users\Daniel\Documents\Arduino\libraries" -fqbn=arduino:avr:uno -ide-version=10609 -build-path "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp" -warnings=none -prefs=build.warn_data_percentage=75 -verbose "C:\3D Printer\eggbot one piece\eggbotaktuell\EggDuino-master\EggDuino\EggDuino.ino"
C:\3D Printer\eggbot one piece\arduino-nightly\arduino-builder -compile -logger=machine -hardware "C:\3D Printer\eggbot one piece\arduino-nightly\hardware" -hardware "C:\Users\Daniel\AppData\Local\Arduino15\packages" -tools "C:\3D Printer\eggbot one piece\arduino-nightly\tools-builder" -tools "C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr" -tools "C:\Users\Daniel\AppData\Local\Arduino15\packages" -built-in-libraries "C:\3D Printer\eggbot one piece\arduino-nightly\libraries" -libraries "C:\Users\Daniel\Documents\Arduino\libraries" -fqbn=arduino:avr:uno -ide-version=10609 -build-path "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp" -warnings=none -prefs=build.warn_data_percentage=75 -verbose "C:\3D Printer\eggbot one piece\eggbotaktuell\EggDuino-master\EggDuino\EggDuino.ino"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\sketch\EggDuino.ino.cpp" -o "nul"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "-IC:\Users\Daniel\Documents\Arduino\libraries\Button" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\sketch\EggDuino.ino.cpp" -o "nul"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "-IC:\Users\Daniel\Documents\Arduino\libraries\Button" "-IC:\Users\Daniel\Documents\Arduino\libraries\AccelStepper" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\sketch\EggDuino.ino.cpp" -o "nul"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "-IC:\Users\Daniel\Documents\Arduino\libraries\VarSpeedServo-master" "-IC:\Users\Daniel\Documents\Arduino\libraries\Button" "-IC:\Users\Daniel\Documents\Arduino\libraries\AccelStepper" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\sketch\EggDuino.ino.cpp" -o "nul"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "-IC:\Users\Daniel\Documents\Arduino\libraries\Button" "-IC:\Users\Daniel\Documents\Arduino\libraries\AccelStepper" "-IC:\Users\Daniel\Documents\Arduino\libraries\VarSpeedServo-master" "-IC:\Users\Daniel\Documents\Arduino\libraries\Arduino-SerialCommand-master" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\sketch\EggDuino.ino.cpp" -o "nul"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "-IC:\Users\Daniel\Documents\Arduino\libraries\Arduino-SerialCommand-master" "-IC:\Users\Daniel\Documents\Arduino\libraries\Button" "-IC:\Users\Daniel\Documents\Arduino\libraries\AccelStepper" "-IC:\Users\Daniel\Documents\Arduino\libraries\VarSpeedServo-master" "C:\Users\Daniel\Documents\Arduino\libraries\Button\Button.cpp" -o "nul"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "-IC:\Users\Daniel\Documents\Arduino\libraries\Button" "-IC:\Users\Daniel\Documents\Arduino\libraries\AccelStepper" "-IC:\Users\Daniel\Documents\Arduino\libraries\VarSpeedServo-master" "-IC:\Users\Daniel\Documents\Arduino\libraries\Arduino-SerialCommand-master" "C:\Users\Daniel\Documents\Arduino\libraries\AccelStepper\AccelStepper.cpp" -o "nul"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "-IC:\Users\Daniel\Documents\Arduino\libraries\Arduino-SerialCommand-master" "-IC:\Users\Daniel\Documents\Arduino\libraries\Button" "-IC:\Users\Daniel\Documents\Arduino\libraries\AccelStepper" "-IC:\Users\Daniel\Documents\Arduino\libraries\VarSpeedServo-master" "C:\Users\Daniel\Documents\Arduino\libraries\AccelStepper\MultiStepper.cpp" -o "nul"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "-IC:\Users\Daniel\Documents\Arduino\libraries\Arduino-SerialCommand-master" "-IC:\Users\Daniel\Documents\Arduino\libraries\Button" "-IC:\Users\Daniel\Documents\Arduino\libraries\AccelStepper" "-IC:\Users\Daniel\Documents\Arduino\libraries\VarSpeedServo-master" "C:\Users\Daniel\Documents\Arduino\libraries\VarSpeedServo-master\VarSpeedServo.cpp" -o "nul"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "-IC:\Users\Daniel\Documents\Arduino\libraries\Button" "-IC:\Users\Daniel\Documents\Arduino\libraries\AccelStepper" "-IC:\Users\Daniel\Documents\Arduino\libraries\VarSpeedServo-master" "-IC:\Users\Daniel\Documents\Arduino\libraries\Arduino-SerialCommand-master" "C:\Users\Daniel\Documents\Arduino\libraries\Arduino-SerialCommand-master\SerialCommand.cpp" -o "nul"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "-IC:\Users\Daniel\Documents\Arduino\libraries\Button" "-IC:\Users\Daniel\Documents\Arduino\libraries\AccelStepper" "-IC:\Users\Daniel\Documents\Arduino\libraries\VarSpeedServo-master" "-IC:\Users\Daniel\Documents\Arduino\libraries\Arduino-SerialCommand-master" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\sketch\EggDuino.ino.cpp" -o "nul"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "-IC:\Users\Daniel\Documents\Arduino\libraries\Button" "-IC:\Users\Daniel\Documents\Arduino\libraries\AccelStepper" "-IC:\Users\Daniel\Documents\Arduino\libraries\VarSpeedServo-master" "-IC:\Users\Daniel\Documents\Arduino\libraries\Arduino-SerialCommand-master" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\sketch\EggDuino.ino.cpp" -o "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\preproc\ctags_target_for_gcc_minus_e.cpp"
"C:\3D Printer\eggbot one piece\arduino-nightly\tools-builder\ctags\5.8-arduino10/ctags" -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\preproc\ctags_target_for_gcc_minus_e.cpp"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -MMD -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "-IC:\Users\Daniel\Documents\Arduino\libraries\Button" "-IC:\Users\Daniel\Documents\Arduino\libraries\AccelStepper" "-IC:\Users\Daniel\Documents\Arduino\libraries\VarSpeedServo-master" "-IC:\Users\Daniel\Documents\Arduino\libraries\Arduino-SerialCommand-master" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\sketch\EggDuino.ino.cpp" -o "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\sketch\EggDuino.ino.cpp.o"
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\libraries\Button\Button.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\libraries\AccelStepper\AccelStepper.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\libraries\AccelStepper\MultiStepper.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\libraries\VarSpeedServo-master\VarSpeedServo.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\libraries\Arduino-SerialCommand-master\SerialCommand.cpp.o
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-gcc" -c -g -x assembler-with-cpp -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10609 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino" "-IC:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\variants\standard" "C:\Users\Daniel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.10\cores\arduino\wiring_pulse.S" -o "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\wiring_pulse.S.o"
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\WInterrupts.c.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\hooks.c.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\wiring.c.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\wiring_analog.c.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\wiring_digital.c.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\wiring_pulse.c.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\wiring_shift.c.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\CDC.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\HardwareSerial.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\HardwareSerial0.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\HardwareSerial1.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\HardwareSerial2.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\HardwareSerial3.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\IPAddress.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\PluggableUSB.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\Print.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\Stream.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\Tone.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\USBCore.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\WMath.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\WString.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\abi.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\main.cpp.o
Using previously compiled file: C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\new.cpp.o
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\wiring_pulse.S.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\WInterrupts.c.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\hooks.c.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\wiring.c.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\wiring_analog.c.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\wiring_digital.c.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\wiring_pulse.c.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\wiring_shift.c.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\CDC.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\HardwareSerial.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\HardwareSerial0.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\HardwareSerial1.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\HardwareSerial2.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\HardwareSerial3.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\IPAddress.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\PluggableUSB.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\Print.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\Stream.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\Tone.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\USBCore.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\WMath.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\WString.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\abi.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\main.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-ar" rcs  "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\core.a" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\core\new.cpp.o"
"C:\3D Printer\eggbot one piece\arduino-nightly\hardware\tools\avr/bin/avr-gcc" -w -Os -Wl,--gc-sections -mmcu=atmega328p  -o "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp/EggDuino.ino.elf" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\sketch\EggDuino.ino.cpp.o" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\libraries\Button\Button.cpp.o" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\libraries\AccelStepper\AccelStepper.cpp.o" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\libraries\AccelStepper\MultiStepper.cpp.o" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\libraries\VarSpeedServo-master\VarSpeedServo.cpp.o" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\libraries\Arduino-SerialCommand-master\SerialCommand.cpp.o" "C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp/core\core.a" "-LC:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp" -lm
C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\sketch\EggDuino.ino.cpp.o: In function `setup':

C:\3D Printer\eggbot one piece\eggbotaktuell\EggDuino-master\EggDuino/EggDuino.ino:131: undefined reference to `makeComInterface()'

C:\3D Printer\eggbot one piece\eggbotaktuell\EggDuino-master\EggDuino/EggDuino.ino:132: undefined reference to `initHardware()'

C:\Users\Daniel\AppData\Local\Temp\build7c2c4feda835a5c1221189ce2d7e36f3.tmp\sketch\EggDuino.ino.cpp.o: In function `loop':

C:\3D Printer\eggbot one piece\eggbotaktuell\EggDuino-master\EggDuino/EggDuino.ino:136: undefined reference to `moveOneStep()'

collect2.exe: error: ld returned 1 exit status

Using library Button at version 1.0.0 in folder: C:\Users\Daniel\Documents\Arduino\libraries\Button 
Using library AccelStepper in folder: C:\Users\Daniel\Documents\Arduino\libraries\AccelStepper (legacy)
Using library VarSpeedServo-master in folder: C:\Users\Daniel\Documents\Arduino\libraries\VarSpeedServo-master (legacy)
Using library Arduino-SerialCommand-master in folder: C:\Users\Daniel\Documents\Arduino\libraries\Arduino-SerialCommand-master (legacy)
exit status 1
Error compiling for board Arduino/Genuino Uno.

--- a/Helper_Functions.ino
+++ b/Helper_Functions.ino
@@ -16,25 +16,25 @@ void initHardware(){
 	penServo.write(penState);
 }
 
-void inline loadPenPosFromEE() {
+inline void loadPenPosFromEE() {
 	penUpPos = eeprom_read_word(penUpPosEEAddress);
 	penDownPos = eeprom_read_word(penDownPosEEAddress);
 	penState = penUpPos;
 }
 
-void inline storePenUpPosInEE() {
+inline void storePenUpPosInEE() {
 	eeprom_update_word(penUpPosEEAddress, penUpPos);
 }
 
-void inline storePenDownPosInEE() {
+inline void storePenDownPosInEE() {
 	eeprom_update_word(penDownPosEEAddress, penDownPos);
 }
 
-void inline sendAck(){
+inline void sendAck(){
 	Serial.print("OK\r\n");
 }
 
-void inline sendError(){
+inline void sendError(){
 	Serial.print("unknown CMD\r\n");
 }
 

